roles/development/files/bin/make-changelog.py

```diff
@@ -106,44 +153,78 @@ def main(username, project, version):
-        commits = [(slug, msg) for slug, msg in commits if pullstr not in msg]
+        if pagure:
+            pullstr = "Merge #"
+
+        commits = [
+            (slug, comdate, msg)
+            for slug, comdate, msg in commits
+            if relstr not in msg]
+        pulls = [
+            (slug, msg)
+            for slug, comdate, msg in commits
+            if pullstr in msg]
+        commits = [
+            (slug, comdate, msg)
+            for slug, comdate, msg in commits
+            if pullstr not in msg]
 
         print
         print start
         print "-" * len(start)
 
+        if pagure:
+            pulls.extend(get_pull_request_pagure(project, commits[0]))
+
         if pulls:
             print
             print "Pull Requests"
             print
 
-        for slug, msg in pulls:
-            number = msg[len(pullstr):].split()[0]
-            try:
-                title, author, link = get_pull_info(username, project, number)
+        if pagure:
+            for number, title, author in pulls:
                 author = "(@%s)" % author
-            except KeyError as e:
-                sys.stderr.write('Problems getting info for '
-                                 '#%s\n%r\n' % (number, e))
-                # Some fallbacks
-                author = ''
-                title = msg
-                link = pull_url_for(username, project, number)
+                link = pull_url_for_pagure(project, number)
 
-            print "- %s #%s, %s\n  %s" % (author.ljust(17), number, title, link)
+                print "- %s #%s, %s\n  %s" % (author.ljust(17), number, title, link)
+
+        else:
+            for slug, msg in pulls:
+                number = msg[len(pullstr):].split()[0]
+                try:
+                    title, author, link = get_pull_info_github(
+                        username, project, number)
+                    author = "(@%s)" % author
+                except KeyError as e:
+                    sys.stderr.write('Problems getting info for '
+                                     '#%s\n%r\n' % (number, e))
+                    # Some fallbacks
+                    author = ''
+                    title = msg
+                    link = pull_url_for_github(username, project, number)
+
+                print "- %s #%s, %s\n  %s" % (author.ljust(17), number, title, link)
 
         if commits:
             print
             print "Commits"
             print
 
-        for slug, msg in commits:
-            print "- %s %s\n  %s" % (
-                slug[:9], msg, commit_url_for(username, project, slug))
+        for slug, comdate, msg in commits:
+            if pagure:
+                print "- %s %s\n  %s" % (
+                    slug[:9], msg, commit_url_for_pagure(project, slug))
+            else:
+                print "- %s %s\n  %s" % (
+                    slug[:9], msg,
+                    commit_url_for_github(username, project, slug))
```

Comment:
Same down here.  We minimize the number of ``if pagure:`` blocks down to just 1 instance if possible.  I don't want to sprinkle all kinds of contingencies throughout.

Perhaps just one lookup at the top:

```python
if pagure:
    commit_url = commit_url_for_pagure
else:
    commit_url = commit_url_for_github
```


And then down here do:
```python
print "- %s %s\n  %s" % (slug[:9], msg, commit_url(username, project, slug))
```